searchlib/src/tests/attribute/posting_store/posting_store_test.cpp

```diff
@@ -178,10 +192,21 @@ PostingStoreTest::test_compact_btree_nodes(uint32_t sequence_length)
     EntryRef old_ref1 = get_posting_ref(1);
     EntryRef old_ref2 = get_posting_ref(2);
     auto usage_before = store.getMemoryUsage();
-    for (uint32_t pass = 0; pass < 15; ++pass) {
-        store.compact_worst_btree_nodes();
+    bool compaction_done = false;
```

Comment:
This code block looks almost identical to the one on lines 162-176. Consider creating a helper function to avoid code duplication.